Sources/Swift/Tools/UIRedactBuilder.swift

```diff
@@ -0,0 +1,136 @@
+        resultRegions.append(CGRect(x: rect.minX,
+                                    y: max(rect.minY, intersectionRect.minY),
+                                    width: intersectionRect.minX - rect.minX,
+                                    height: min(intersectionRect.maxY, rect.maxY) - max(rect.minY, intersectionRect.minY)))
+        
+        // Calculate the right region.
+        resultRegions.append(CGRect(x: intersectionRect.maxX,
+                                    y: max(rect.minY, intersectionRect.minY),
+                                    width: rect.maxX - intersectionRect.maxX,
+                                    height: min(intersectionRect.maxY, rect.maxY) - max(rect.minY, intersectionRect.minY)))
+        
+        return resultRegions.filter { !$0.isEmpty }.map { RedactRegion(rect: $0, color: color) }
+    }
+}
+
+class UIRedactBuilder {
+    
+    //This is a list of UIView subclasses that will be ignored during redact process
+    var ignoreClasses: [AnyClass] 
+    //This is a list of UIView subclasses that need to be redacted from screenshot
+    var redactClasses: [AnyClass]
+    
+    init() {
+        redactClasses = [ UILabel.self, UITextView.self, UITextField.self ] + [
+            "_TtCOCV7SwiftUI11DisplayList11ViewUpdater8Platform13CGDrawingView",
+            "_TtC7SwiftUIP33_A34643117F00277B93DEBAB70EC0697122_UIShapeHitTestingView",
+            "SwiftUI._UIGraphicsView", "SwiftUI.ImageLayer"
+        ].compactMap { NSClassFromString($0) }
+#if os(iOS)
+        ignoreClasses = [ UISlider.self, UISwitch.self ]
+#else
+        ignoreClasses = []
+#endif
+    }
+    
+    func redactRegionsFor(view: UIView, options: SentryRedactOptions?) -> [RedactRegion] {
+        var redactingRegions = [RedactRegion]()
+        
+        self.mapRedactRegion(fromView: view,
+                             to: view,
+                             redacting: &redactingRegions,
+                             area: view.frame,
+                             redactText: options?.redactAllText ?? true,
+                             redactImage: options?.redactAllImages ?? true)
+        
+        return redactingRegions
+    }
+    
+    private func shouldIgnore(view: UIView) -> Bool {
+        ignoreClasses.contains { view.isKind(of: $0) }
+    }
+    
+    private func shouldRedact(view: UIView, redactText: Bool, redactImage: Bool) -> Bool {
+        if redactImage, let imageView = view as? UIImageView {
+            return shouldRedact(imageView: imageView)
+        }
+        return redactText && redactClasses.contains { view.isKind(of: $0) }
+    }
+    
+    private func shouldRedact(imageView: UIImageView) -> Bool {
+        // Checking the size is to avoid redact gradient backgroud that
+        // are usually small lines repeating
+        guard let image = imageView.image, image.size.width > 10 && image.size.height > 10  else { return false }
+        return image.imageAsset?.value(forKey: "_containingBundle") == nil
```

Comment:
`l`: Out of curiosity. What is the `_containingBundle`?

Reply:
Information about the bundle that the image is coming from.